tooling/camel-karaf-feature-maven-plugin/src/main/java/org/apache/camel/karaf/feature/maven/EnsureWrapBundleNameMojo.java

```diff
@@ -0,0 +1,144 @@
+        StringBuilder result = new StringBuilder();
+
+        // Capitalize first letter of each word
+        for (int i = 0; i < words.length; i++) {
+            String word = words[i];
+            if (word.isEmpty()) {
+                continue;
+            }
+            result.append(Character.toUpperCase(word.charAt(0)));
+            if (word.length() > 1) {
+                result.append(word.substring(1).toLowerCase());
+            }
+            // Only add %20 if it's not the last word
+            if (i < words.length - 1) {
+                result.append("%20");
+            }
+        }
+        return result.toString();
+    }
+
+    @Override
+    protected boolean processWrappedBundle(WrappedBundle wrappedBundle) {
+        Bundle bundle = wrappedBundle.getBundle();
+        String location = bundle.getLocation();
+        try {
+            bundle.setLocation(processLocation(wrappedBundle));
+        } catch (Exception e) {
+            getLog().error("Could not process the Bundle location '%s': %s".formatted(location, e.getMessage()), e);
+        }
+        return false;
+    }
+
+    String processLocation(WrappedBundle wrappedBundle) throws Exception {
+        String location = wrappedBundle.getBundle().getLocation();
+        String instructions = wrappedBundle.getInstructions();
+        boolean dollarNeeded = !(instructions != null && instructions.contains("$"));
+
+        String bundleNameHeader = "%s=Wrap%%20of%%20%s".formatted(BUNDLE_NAME, toPascalCase(wrappedBundle.getArtifactId()));
+        String bundleSymbolicNameHeader = "%s=wrap_%s.%s".formatted(BUNDLE_SYMBOLIC_NAME, wrappedBundle.getGroupId(), wrappedBundle.getArtifactId());
+        Map<String, String> headers = new TreeMap<>(); //sorted by key
+        headers.put(BUNDLE_NAME, bundleNameHeader);
+        headers.put(BUNDLE_SYMBOLIC_NAME, bundleSymbolicNameHeader);
+
+        return insertHeaderIfNeeded(dollarNeeded, location, headers);
+    }
+
+
+    private String insertHeaderIfNeeded(boolean dollarNeeded, String location, Map<String, String> headers) {
+
+        StringBuilder sb = new StringBuilder(location);
+        for (Map.Entry<String, String> entry : headers.entrySet()) {
+            if (location.contains(entry.getKey())) {
+                continue;
+            }
+            boolean hasChanged = false;
+            // insert before existing headers header
+            for (String header : HEADERS_AFTER_BUNDLE_NAME) {
+                // add Bundle-Version before
+                if (location.contains(header)) {
+                    int versionHeaderStartIndex = location.indexOf(header);
+                    if (dollarNeeded) {
+                        // "amp;" is automatically added
+                        dollarNeeded = false;
+                        location = sb.insert(versionHeaderStartIndex, "$%s&".formatted(entry.getValue())).toString();
+                        hasChanged = true;
+                        break;
```

Comment:
You seem to repeat the exact same lines in both cases (if and else blocks) so I think that you can move them after the if/else block